src/figure/movement.c

```diff
@@ -373,13 +373,21 @@ void figure_movement_move_ticks(figure *f, int num_ticks)
     walk_ticks(f, num_ticks, 0);
 }
 
-void figure_movement_move_ticks_with_percentage(figure* f, int num_ticks, int tick_percentage)
+void figure_movement_move_ticks_with_percentage(figure* f, int num_ticks, int tick_percentage, int slow_percentage)
```

Comment:
Instead of having a new `slow_percentage` (which is cumbersome as the figures can be allowed to move both faster and slower, why not simply slow down movement if `tick_percentage` is negative and reaches `-100`?